### ISSUE TITLE
pathfinding: unnecessary error

### ISSUE THREAD

**Issue by tenplus1** [minetest/minetest#6771]
Sometimes when using pathfinding it comes up an unnecessary error which I feel should be removed since the command itself either returns a table path or nil.

e.g.

2017-12-11 18:52:16: WARNING[Server]: Pathfinder: failed to update cost map
2017-12-11 18:52:16: WARNING[Server]: Pathfinder: failed to update cost map
2017-12-11 18:52:18: WARNING[Server]: Pathfinder: failed to update cost map
2017-12-11 18:52:18: WARNING[Server]: Pathfinder: failed to update cost map
2017-12-11 18:52:21: WARNING[Server]: Pathfinder: failed to update cost map
2017-12-11 18:52:22: WARNING[Server]: Pathfinder: failed to update cost map
2017-12-11 18:52:23: WARNING[Server]: Pathfinder: failed to update cost map
2017-12-11 18:52:24: WARNING[Server]: Pathfinder: failed to update cost map
2017-12-11 18:52:25: WARNING[Server]: Pathfinder: failed to update cost map

**Comment by IhrFussel** [comment]
I agree, it can be logged quite a lot in the debug.txt in a matter of minutes.

Also it seems to cause some lag with all those warnings displaying, but this is not confirmed yet.

**Comment by paramat** [comment]
We can move it to 'infostream', easily done and i support.

**Comment by tenplus1** [comment]
So long as it doesn't appear in chat or console that's a good idea paramat :)

**Comment by paramat** [comment]
Correct, only in debug if log level is 'info' or above.
Easy PR to make, anyone?

**Comment by rubenwardy** [comment]
It's probably better to fix pathfinding than to hide errors

**Comment by paramat** [comment]
I got the impression from comments it's not an error, i have no idea if it is or not, but yes the cause should be investigated.

**Comment by JDCodeIt** [comment]
I've had a look at the code, but don't quite understand it totally. It seems the "false" return value from Pathfinder::updateAllCosts could occur under condition where there is already a shorter path found - why is this considered an error? Once that happens, the path is not returned, so this return condition while updating the cost map results in no path being returned to the original caller.

**Comment by paramat** [comment]
it seems this 'error' is shown when pathfinding is functioning reasonably, so it really shouldn't be an 'error'. We can make a PR and even keep the issue open to investigate if this is actually is an error or not.
1 line PR i will merge as trivial.

**Comment by paramat** [comment]
https://github.com/minetest/minetest/blob/1992db1395d9c068327a7c08bac7a24ef7112274/src/pathfinder.cpp#L736
->
`INFO_TARGET << "failed to update cost map"<< std::endl;`

**Comment by JDCodeIt** [comment]
I think the "error" is that the cost map generation code is returning "false" under normal conditions, and this will prevent a path from being calculated. When you see this error in the console - your path did not get created.
Some testing I did showed that all 3 algorithms fail to find paths even in some of the most obvious circumstances - perhaps only because the cost map calculations are returning false.

If I'm reading the cost calculator correctly it was going to check each of the 4 x/z possible starting directions - then if during that search the previous direction's cost was lower, it bails out and returns false - seem wrong.

**Comment by JDCodeIt** [comment]
Emperical results with stable 0.4.17 - replacing pathfinder.cpp:1008	<code>return false;</code> with <code>continue;</code> yields a path for almost all cases - but sometimes an ugly one.
Similar logic is on line 1169 of the same file for the "A*" algorithms

**Comment by tenplus1** [comment]
Most of the errors in my case were caused due to path limitations (say 30 nodes long) and not being able to find a route to the player because of decoration or high steps, so in this case it should return nil without the error.

**Comment by rubenwardy** [comment]
This error no longer exists
